src/poetry/console/commands/check.py

```diff
@@ -19,6 +25,32 @@ def handle(self) -> int:
         poetry_file = Factory.locate(Path.cwd())
         config = PyProjectTOML(poetry_file).poetry_config
         check_result = Factory.validate(config, strict=True)
+
+        unrecognized = sorted(
+            set(config.get("classifiers", []))
+            - set(classifiers)
+            - set(deprecated_classifiers)
+        )
+        if unrecognized:
+            check_result["errors"].append(
+                f"Unrecognized classifiers: {unrecognized!r}."
+            )
+
+        deprecated = sorted(
+            set(config.get("classifiers", [])).intersection(set(deprecated_classifiers))
+        )
+        if deprecated:
+            for old_classifier in deprecated:
+                message = f"Deprecated classifier {old_classifier!r}."
```

Comment:
Does it make more sense to print the repr (`!r`) here than the string form, and if so, why?

Reply:
The fully-qualified classifier is a string delimited by ` :: ` separators. To make the error message more readable I wanted to have visual clues to materialize the start and end of the classifier string. That way the user can easely copy and paste the classifier string from the error message while reducing mistakes pertaining to extra spaces from the full error message.

I chose `!r` for consistency with the `{new_classifiers!r}` message below, which is a list of classifiers. That way I'm certain all classifiers strings will be rendered with the same kind of quotes (i.e. simple tick: `'`).

Do you want me to add that explanation as comment in the source code?

Reply:
I added that explanation into the classifier helper docstring at: https://github.com/python-poetry/poetry/pull/2881/commits/d1f7e5ca2c5734a29ad022475d1f3baa30a6e5be